### ISSUE TITLE
Disperancies between validation scores and test scores

### ISSUE THREAD

**Issue by btseytlin** [KevinMusgrave/powerful-benchmarker#58]
I am running some benchmarks and I notice that for all methods (that I tried) the validation MAPR is around 0.4, whilist test MAPR is usually around 0.25 (CUB dataset).

I see the following explanations:
* The CV pipeline is not reflective of the real train/test split, so it's easier for the model to beat validation.
* The validation set is smaller, so models perform way better on it. Perhaps less hard examples make it into validation? Class imbalance issues?
* Distribution shift between train and test, since train and test have non-overlapping classes. If this is the case then there is no way to avoid it.

@KevinMusgrave how do you think, what is happening here?

**Comment by KevinMusgrave** [comment]
I believe the main cause is that the validation set is smaller. Using the default split settings on CUB: 

- the test set has 100 classes
- each train/val split consists of 75 train and 25 val classes

Because there are fewer classes in the validation set, the embedding space is "less crowded", so retrieval becomes easier.

**Comment by btseytlin** [comment]
Thanks!

In that case, is it a good idea to optimize hyperparameters on such validation sets?

**Comment by KevinMusgrave** [comment]
If we assume that the test set is the real-world data, then we can't touch that. Therefore we can only deal with the remaining 100 classes. Increasing the size of the validation set will make it more similar to the test set, but then each cross-validated model sees fewer samples during training. It's a trade-off, and there might be a better train/val split than 75/25. But I think it's best to maximize the number of training set classes, while still having a useful validation set.
